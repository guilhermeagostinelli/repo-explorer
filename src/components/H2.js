import styled from "styled-components";

const H2 = styled.h2`
  font-size: 1em;
  color: ${props => props.color || "black"};
`;

export default H2;
