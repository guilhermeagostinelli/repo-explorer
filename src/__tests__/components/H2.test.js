import React from "react";
import { create } from "react-test-renderer";
import H2 from "../../components/H2";
import "jest-styled-components";

describe("<H2 />", () => {
  it("should match the snapshot when no color is specified", () => {
    const h2 = create(<H2 />);
    expect(h2.toJSON()).toMatchSnapshot();
  });

  it("should match the snapshot when a color is specified", () => {
    const h2 = create(<H2 color="blue" />);
    expect(h2.toJSON()).toMatchSnapshot();
  });
});
