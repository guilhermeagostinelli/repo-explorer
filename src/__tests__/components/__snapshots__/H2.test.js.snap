// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H2 /> should match the snapshot when a color is specified 1`] = `
.c0 {
  font-size: 1em;
  color: blue;
}

<h2
  className="c0"
  color="blue"
/>
`;

exports[`<H2 /> should match the snapshot when no color is specified 1`] = `
.c0 {
  font-size: 1em;
  color: black;
}

<h2
  className="c0"
/>
`;
